### ISSUE TITLE
SBT plugin SNAPSHOT not updated

### ISSUE THREAD

**Issue by galderz** [sbt/sbt#646]
My SBT plugin SNAPSHOT does not seem to be get updated with SBT 0.12.1:

I've pushed a new update to:
https://repository.jboss.org/nexus/content/groups/public/io/escalante/sbt/sbt-escalante_2.9.2_0.12/0.1.0-SNAPSHOT/

But, the debug log of update (after doing: reload plugins) shows:
https://gist.github.com/4563950

I have resolvers set in the project/build.sbt as:

```
resolvers ++= Seq(
   "JBoss repository" at "http://repository.jboss.org/nexus/content/groups/public/",
   "Project Odd repository" at "http://repository-projectodd.forge.cloudbees.com/upstream/"
)

addSbtPlugin("io.escalante.sbt" % "sbt-escalante" % "0.1.0-SNAPSHOT")
```

Why are the resolvers I've set in `project/build.sbt` not checked?

Locally, .ivy2/cache/scala_2.9.2/sbt_0.12/io.escalante.sbt/sbt-escalante/ivy-0.1.0-SNAPSHOT.xml shows: https://gist.github.com/4563954

**Comment by galderz** [comment]
As a side note, if I face similar problems, how do you manually remove the SNAPSHOT dependency in a safe way without nuking `.ivy2/local` and `ivy2/cache` ?

**Comment by harrah** [comment]
Please see http://www.scala-sbt.org/release/docs/Detailed-Topics/Dependency-Management-Flow.html#plugins.  Plugins are not handled by the normal `update`.

You can be more fine-grained in your cache clearing by removing the specific directories under ~/.ivy2/local and ~/.ivy2/cache, such as ~/.ivy2/cache/org.example/demo/1.0/.  Also, note that there was a cache bug related to the local repository that got fixed in 0.12.2.
